### PR TITLE
Append ignored xUnit2 ITestOutputHelper messages

### PR DESCRIPTION
**Pull request by aptester**
[//]: # (
. Thank you so much for sending us a pull request! 
.
. Make sure you have a clear name for your pull request. 
. The name should start with a capital letter and no dot is required in the end of the sentence.
. To link the request with isses use the following notation: (fixes #123, fixes #321\)
.
. An example of good pull request names:
. - Add Russian translation (fixes #123\)
. - Add an ability to disable default plugins
. - Support emoji in test descriptions
)

### Context
[//]: # (
Describe the problem or feature in addition to a link to the issues
)
Output captured by using xUnit2 ITestOutputHelper class (https://xunit.github.io/docs/capturing-output.html), is being ignored when Allure report is being generated. Attached is a sample xUnit2 test result xml which has user defined output. 
[xunit-result.txt](https://github.com/allure-framework/allure2/files/1106367/xunit-result.txt)

The sample xunit2 result xml provided in #139 did not have any <output> tag. This PR appends output to the end of the test's status details' 'message'. 

#### Checklist
- [x] [Sign Allure CLA][cla]
- [x] Provide unit tests

[cla]: https://cla-assistant.io/accept/allure-framework/allure2

### CODE REVIEW

**Comment by baev** [comment]
Thanks for the contribution! Could you please sign CLA and fix the tests, and then I'll review and merge

**Comment by aptester** [comment]
reopening PR

**Comment by aptester** [comment]
@baev how can I see the failure report : file:///home/jenkins/workspace/allure2_PR-433-CWLLC2UZ6WPN65GET6IWZED3LS4UX7QJYHFQX66TUQZD2T4SC7NQ/plugins/xunit-xml-plugin/build/reports/tests/test/index.html?

**Comment by sseliverstov** [comment]
https://ci.qameta.io/blue/organizations/jenkins/allure2/detail/PR-433/2/pipeline

**Comment by baev** [comment]
```
io.qameta.allure.xunitxml.XunitXmlPluginTest > shouldSetStatusDetails FAILED
    org.junit.experimental.theories.internal.ParameterizedAssertionError
        Caused by: java.lang.AssertionError at XunitXmlPluginTest.java:155

5 tests completed, 1 failed
:xunit-xml-plugin:test FAILED

```

**Comment by aptester** [comment]
The '.\gradlew build' command is passing without any failed unit tests on my laptop. Any idea why that may be happening?

**Comment by baev** [comment]
Just cloned your fork, and 1 test is failed with the following error:

```
org.junit.experimental.theories.internal.ParameterizedAssertionError: shouldSetStatusDetails("[Ljava.lang.String;@5de6e275" <from input[0]>)
	at org.junit.experimental.theories.Theories$TheoryAnchor.reportParameterizedError(Theories.java:288)
	at org.junit.experimental.theories.Theories$TheoryAnchor$1$1.evaluate(Theories.java:237)
	at org.junit.experimental.theories.Theories$TheoryAnchor.runWithCompleteAssignment(Theories.java:218)
	at org.junit.experimental.theories.Theories$TheoryAnchor.runWithAssignment(Theories.java:204)
	at org.junit.experimental.theories.Theories$TheoryAnchor.runWithIncompleteAssignment(Theories.java:212)
	at org.junit.experimental.theories.Theories$TheoryAnchor.runWithAssignment(Theories.java:202)
	at org.junit.experimental.theories.Theories$TheoryAnchor.evaluate(Theories.java:187)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecuter.runTestClass(JUnitTestClassExecuter.java:114)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecuter.execute(JUnitTestClassExecuter.java:57)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassProcessor.processTestClass(JUnitTestClassProcessor.java:66)
	at org.gradle.api.internal.tasks.testing.SuiteTestClassProcessor.processTestClass(SuiteTestClassProcessor.java:51)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:35)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.dispatch.ContextClassLoaderDispatch.dispatch(ContextClassLoaderDispatch.java:32)
	at org.gradle.internal.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:93)
	at com.sun.proxy.$Proxy1.processTestClass(Unknown Source)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.processTestClass(TestWorker.java:109)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:35)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:146)
	at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:128)
	at org.gradle.internal.remote.internal.hub.MessageHub$Handler.run(MessageHub.java:404)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:63)
	at org.gradle.internal.concurrent.StoppableExecutorImpl$1.run(StoppableExecutorImpl.java:46)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:55)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.AssertionError: 
Expecting:
  <[("Assert.True() Failure\r\nExpected: True\r\nActual:   False
test output
", "FAILED-TRACE")]>
to contain exactly in any order:
  <[("Assert.True() Failure\r\nExpected: True\r\nActual:   False
test output
", "FAILED-TRACE")]>
elements not found:
  <[("Assert.True() Failure\r\nExpected: True\r\nActual:   False
test output
", "FAILED-TRACE")]>
and elements not expected:
  <[("Assert.True() Failure\r\nExpected: True\r\nActual:   False
test output
", "FAILED-TRACE")]>

	at io.qameta.allure.xunitxml.XunitXmlPluginTest.shouldSetStatusDetails(XunitXmlPluginTest.java:155)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.experimental.theories.Theories$TheoryAnchor$2.evaluate(Theories.java:274)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:48)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.experimental.theories.Theories$TheoryAnchor$1$1.evaluate(Theories.java:232)
	... 43 more
```

**Comment by aptester** [comment]
Thanks @baev! Looks like the newline is causing trouble. Probably due to the fact I'm using Windows. I'll send out an updated PR shortly.

**Comment by aptester** [comment]
@baev please review

File: plugins/xunit-xml-plugin/src/main/java/io/qameta/allure/xunitxml/XunitXmlPlugin.java [hunk 1]
```diff
@@ -154,6 +155,20 @@ private Status getStatus(final XmlElement testElement) {
                             .ifPresent(details::setTrace);
                     return details;
                 });
+
+        Optional<String> output = testElement.getFirst(OUTPUT_ELEMENT_NAME)
+                                             .map(XmlElement::getValue);
+        return output.isPresent() ? addOutputToStatusDetails(statusDetails, output) : statusDetails;
+    }
+
+    private Optional<StatusDetails> addOutputToStatusDetails(final Optional<StatusDetails> statusDetails,
```
Comment (baev) [inline plugins/xunit-xml-plugin/src/main/java/io/qameta/allure/xunitxml/XunitXmlPlugin.java]: do not use `Optional` as method arguments

File: plugins/xunit-xml-plugin/src/main/java/io/qameta/allure/xunitxml/XunitXmlPlugin.java [hunk 2]
```diff
@@ -154,6 +155,20 @@ private Status getStatus(final XmlElement testElement) {
...
+                                                             final Optional<String> output) {
+        StringBuilder message = new StringBuilder();
+        statusDetails.ifPresent(s -> message.append(s.getMessage()).append(System.getProperty("line.separator")));
```
Comment (aptester) [inline plugins/xunit-xml-plugin/src/main/java/io/qameta/allure/xunitxml/XunitXmlPlugin.java]: I wanted to separate message and output values by a 'newline' character.

Comment (baev) [inline plugins/xunit-xml-plugin/src/main/java/io/qameta/allure/xunitxml/XunitXmlPlugin.java]: simply use `\n`, it should be processed correctly

**Comment by baev** [comment]
@aptester could you please fix one more warning 

```
[Whitespace | WhitespaceAround] io.qameta.allure.xunitxml.(XunitXmlPlugin.java:169)
  WhitespaceAround: 'if' is not followed by whitespace. Empty blocks may only be represented as {} when not part of a multi-block statement (4.1.3)
  http://checkstyle.sourceforge.net/config_whitespace.html#WhitespaceAround
```

we are going to release 2.2.1 today, so you can get your change included

**Comment by baev** [comment]
@aptester thanks!

**Pull request merged**